debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/rest/DebeziumMongoDbConnectorResource.java

```diff
@@ -41,6 +59,48 @@ public Connector getConnector() {
         return new MongoDbConnector();
     }
 
+    protected MongoDbConnection primary(MongoDbTaskContext context) throws DebeziumException {
+        ReplicaSet replicaSet = new ReplicaSet(context.getConnectionContext().connectionString());
+        return context.getConnectionContext().connect(replicaSet, context.filters(), (s, throwable) -> {
+            throw new DebeziumException(s, throwable);
+        });
+    }
+
+    protected <T extends DataCollectionId> Stream<T> determineDataCollectionsToBeSnapshotted(
+                                                                                             CommonConnectorConfig connectorConfig,
+                                                                                             final Collection<T> allDataCollections) {
+        final Set<Pattern> snapshotAllowedDataCollections = connectorConfig.getDataCollectionsToBeSnapshotted()
+                .stream()
+                .map(regex -> Pattern.compile(regex, Pattern.CASE_INSENSITIVE))
+                .collect(Collectors.toSet());
+        if (snapshotAllowedDataCollections.size() == 0) {
+            return allDataCollections.stream();
+        }
+        else {
+            return allDataCollections.stream()
+                    .filter(dataCollectionId -> snapshotAllowedDataCollections.stream()
+                            .anyMatch(s -> s.matcher(dataCollectionId.identifier()).matches()));
+        }
+    }
+
+    @Override
+    public List<DataCollection> getMatchingCollections(Configuration configuration) {
+        MongoDbTaskContext context = new MongoDbTaskContext(configuration);
```

Comment:
Please avoid naming variables "context". I know it is still the case in test sources, however I've purposefully tried eliminating this in the primary code in order to make it more visible which context the variable holds reference to (we have too many damn contexts).

Reply:
Maybe this detail level is not what we should be concerned about in PRs as long as var names are totally off, no? I got your point and if that method would be longer or having multiple context vars I fully agree and would have named it less generic, sure.